source/mesh_refinement/composition_gradient.cc

```diff
@@ -40,41 +41,48 @@ namespace aspect
       Vector<float> this_indicator (indicators.size());
       const double power = 1.0 + dim/2.0;
 
-      const Quadrature<dim> quadrature(this->get_fe().base_element(this->introspection().base_elements.compositional_fields).get_unit_support_points());
-      FEValues<dim> fe_values (this->get_mapping(),
-                               this->get_fe(),
-                               quadrature,
-                               update_quadrature_points | update_gradients);
-
-      // the values of the compositional fields are stored as block vectors for each field
-      // we have to extract them in this structure
-      std::vector<Tensor<1,dim>> composition_gradients (quadrature.size());
-
-      for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+      for (const unsigned int k : this->introspection().get_compositional_field_base_element_indices())
```

Comment:
This is a long loop body to carry around a single-character variable name that is not standard (like `c` for composition). Perhaps name it `base_element_index` or some such?